src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TakeBlockOnActivationSystem.java

```diff
@@ -61,16 +62,17 @@ public void onActivate(ActivateEvent event, EntityRef entity) {
         BlockComponent blockComponent = entity.getComponent(BlockComponent.class);
         LASTeamComponent flagTeamComponent = entity.getComponent(LASTeamComponent.class);
 
-
         EntityRef flagTaker = event.getInstigator();
         LASTeamComponent playerTeamComponent = flagTaker.getComponent(LASTeamComponent.class);
         // If the flag being taken is a red flag and the player is on the black team, let them take the flag
         if (flagTeamComponent.team.equals(LASUtils.RED_TEAM) && playerTeamComponent.team.equals(LASUtils.BLACK_TEAM)) {
+            flagTaker.addComponent(new HasFlagComponent(LASUtils.RED_TEAM));
             inventoryManager.giveItem(flagTaker, EntityRef.NULL, blockFactory.newInstance(blockManager.getBlockFamily(LASUtils.RED_FLAG_URI)));
             worldProvider.setBlock(blockComponent.getPosition(), blockManager.getBlock(BlockManager.AIR_ID));
             entity.destroy();
         }
         if (flagTeamComponent.team.equals(LASUtils.BLACK_TEAM) && playerTeamComponent.team.equals(LASUtils.RED_TEAM)) {
+            flagTaker.addComponent(new HasFlagComponent(LASUtils.BLACK_TEAM));
             inventoryManager.giveItem(flagTaker, EntityRef.NULL, blockFactory.newInstance(blockManager.getBlockFamily(LASUtils.BLACK_FLAG_URI)));
             worldProvider.setBlock(blockComponent.getPosition(), blockManager.getBlock(BlockManager.AIR_ID));
             entity.destroy();
```

Comment:
I know the above is not strictly a part of this PR, but I'd love it if you added a different method inside the onActivate with an intuitive name.
Also the above if condition has duplicate code, so maybe it can be extracted to a separate method that gets called with Black/Red team.